src/rust/es.tumangaonline/src/lib.rs

```diff
@@ -0,0 +1,386 @@
+	let author = title_elements
+		.first()
+		.attr("title")
+		.read()
+		.replace(", ", "");
+	let artist = title_elements.last().attr("title").read().replace(", ", "");
+	let description = html.select("p.element-description").text().read();
+
+	let categories = html
+		.select("a.py-2")
+		.array()
+		.map(|x| {
+			x.as_node()
+				.expect("node array element should be a node")
+				.text()
+				.read()
+		})
+		.collect::<Vec<String>>();
+
+	let status_text = html.select("span.book-status").text().read();
+	let status = match status_text.as_str() {
+		"Publicándose" => MangaStatus::Ongoing,
+		"Finalizado" => MangaStatus::Completed,
+		_ => MangaStatus::Unknown,
+	};
+
+	let nsfw = if !html.select("i.fa-heartbeat").array().is_empty() {
+		MangaContentRating::Nsfw
+	} else if categories.iter().any(|x| x == "Ecchi") {
+		MangaContentRating::Suggestive
+	} else {
+		MangaContentRating::Safe
+	};
+
+	let type_text = html.select("h1.book-type").text().read();
+	let viewer = match type_text.as_str() {
+		"MANHWA" => MangaViewer::Scroll,
+		"MANHUA" => MangaViewer::Scroll,
+		_ => MangaViewer::Rtl,
+	};
+
+	Ok(Manga {
+		id,
+		cover,
+		title,
+		author,
+		artist,
+		description,
+		url,
+		categories,
+		status,
+		nsfw,
+		viewer,
+	})
+}
+
+fn parse_chapter(element: Node) -> Chapter {
+	let url = element
+		.select("div.row > .text-right > a")
+		.attr("href")
+		.read();
+	let id = url.strip_prefix(BASE_URL).unwrap_or(&url).to_owned();
+
+	let scanlator = element.select("div.col-md-6.text-truncate").text().read();
+
+	let date_updated = element
+		.select("span.badge.badge-primary.p-2")
+		.first()
+		.text()
+		.as_date("yyyy-MM-dd", None, None);
+
+	Chapter {
+		id,
+		url,
+		scanlator,
+		date_updated,
+		..Default::default()
+	}
+}
+
+#[get_chapter_list]
+fn get_chapter_list(id: String) -> Result<Vec<Chapter>> {
+	let url = if id.starts_with("http") {
+		id
+	} else {
+		format!("{BASE_URL}{id}")
+	};
+	let html = Request::new(&url, HttpMethod::Get)
+		.header("User-Agent", USER_AGENT)
+		.header("Referer", BASE_URL)
+		.html()?;
+
+	let chapter_elements = html.select("div.chapters > ul.list-group li.p-0.list-group-item");
+
+	let mut chapters: Vec<Chapter> = Vec::new();
+
+	if chapter_elements.array().is_empty() {
+		// one shot
+		let elements = html
+			.select("div.chapter-list-element > ul.list-group li.list-group-item")
+			.array();
+		for element in elements {
+			let mut chapter = parse_chapter(
+				element
+					.as_node()
+					.expect("html array element should be a node"),
+			);
+			chapter.title = String::from("One Shot");
+			chapters.push(chapter);
+		}
+	} else {
+		for element in chapter_elements.array() {
+			let element = element
+				.as_node()
+				.expect("html array element should be a node");
+			let title = element.select("div.col-10.text-truncate").text().read();
+			let num_text = element.select("a.btn-collapse").text().read();
+			let chapter_num = helper::string_between(&num_text, "Capítulo ", ":", 0)
```

Comment:
could probably replace this with [`aidoku::helpers::substring::Substring`](https://aidoku.github.io/aidoku-rs/aidoku_helpers/substring/trait.Substring.html)

requires enabling the `helpers` feature ofc

Reply:
I was thinking about that but I thought the string_between was cleaner since I wouldn't have to deal with optional unwrapping, but ig I could try it

Reply:
probably should've made it return a blank string on failure like in kotlin lol